core/store/mmap_directory.cpp

```diff
@@ -104,6 +104,40 @@ std::shared_ptr<mmap_handle> OpenHandle(const std::filesystem::path& dir,
   return nullptr;
 }
 
+#ifdef __linux__
+IRS_NO_INLINE size_t BytesInCache(uint8_t* addr, size_t length) {
+  static const size_t kPageSize = sysconf(_SC_PAGESIZE);
+  IRS_ASSERT(reinterpret_cast<uintptr_t>(addr) % kPageSize == 0);
+  std::vector<uint8_t> pages(
```

Comment:
I suggest to use `std::array` instead

Reply:
vector is better for small mapping, especially if we will make bigger max size